dev/tools/dartdoc_checker.dart

```diff
@@ -81,12 +79,12 @@ void checkForUnresolvedDirectives(String htmlOutputPath) {
 
 int _scanFile(File file) {
   assert(path.extension(file.path) == '.html');
-  final Iterable<String> matches = _pattern.allMatches(file.readAsStringSync())
+  final Iterable<String?> matches = _pattern.allMatches(file.readAsStringSync())
       .map((RegExpMatch m ) => m.group(0));
```

Comment:
I wonder if this would be better expressed as

```suggestion
  final Iterable<String> matches = _pattern.allMatches(file.readAsStringSync())
      .map((RegExpMatch m ) => m.group(0)!);
```